File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        query = {}  # Hash map for storing remainder and key pair
        for i in range(len(nums)):
            if nums[i] in query:
                return [query[nums[i]],i]   # if remainder is same as current scan we can return 
            else:
                query[target-nums[i]]=i;    # storing remainder with key
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n= nums.size();
        unordered_map<int,int> a;
        for(int i=0; i<n; i++){
            auto it= a.find(target-nums[i]);
            if(it!=a.end()){
                return {i,it->second};
            }else{
                a.insert({nums[i],i});
            }
            
        }
        vector<int> ret1(2,0);
        return ret1;
    }
};